FE/airbnb-fe/src/components/Atoms/Search/Check/CalendarCarousel/SingleCalendar.jsx

```diff
@@ -0,0 +1,194 @@
+import React, { useContext } from 'react';
+import styled from 'styled-components';
+import { SearchContext } from '../..';
+//달력 생성
+const createMonthArray = (year, month) => {
+  let monthArr = [];
+  let weekArr = [];
+  let dayCnt = 1;
+
+  const firstDayName = new Date(year, month - 1, 1).getDay(); //6
+  //이번달 마지막 날짜
+  const lastDay = new Date(year, month, 0).getDate();
+  // i: 1주 j:요일
+  for (let i = 0; i < 7; i++) {
+    for (let j = 0; j < 7; j++) {
+      if (weekArr.length >= 7) {
+        monthArr.push(weekArr);
+        weekArr = [];
+      }
+      // 이번달 시작 요일 전
+      if (i === 0 && j < firstDayName) weekArr.push(' ');
+      //이번달 시작 요일 주
+      else if (i === 0 && j >= firstDayName) {
+        weekArr.push(dayCnt);
+        dayCnt++;
+      }
+      //이번달의 마지막날, 혹은 그 이전일 때
+      else if (i > 0 && dayCnt <= lastDay) {
+        weekArr.push(dayCnt);
+        dayCnt++;
+      } else if (dayCnt > lastDay && weekArr.length > 0) weekArr.push(' ');
+    }
+  }
+  return monthArr;
+};
+
+const SingleCalendar = ({ range }) => {
+  const { calendarData, calDispatch } = useContext(SearchContext);
+  const today = new Date(Date.now());
+  const [year, month] = [today.getFullYear(), range + today.getMonth() + 1];
+  const monthArr = createMonthArray(year, month);
+  // console.log('달', monthArr);
+
+  const handleDisabled = (day) => {
+    let tempDate = new Date(year, month - 1, day, 23, 59, 59);
+    return today > tempDate;
+  };
+
+  const handleTdBtnClick = (e) => {
+    console.log(e.target.dataset.day);
+    const dateArr = e.target.dataset.day.split('-').map(Number); // "2021-5-26"
```

Comment:
이 시점에서 year, month, day 로 배열에 대한 구조 분해 할당을 해주어도 될 것 같습니다!

만약 closure에 있는 변수명과 겹쳐서 고민이셨다면 다른 이름을 사용하거나 또는 겹쳐도 괜찮지 않을까 싶습니다.

```suggestion
    const [year, month, day] = e.target.dataset.day.split('-').map(Number); // "2021-5-26"
```